aws/resource_aws_secretsmanager_secret_rotation_test.go

```diff
@@ -0,0 +1,169 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccAwsSecretsManagerSecretRotation(t *testing.T) {
+	var secret secretsmanager.DescribeSecretOutput
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_secretsmanager_secret_rotation.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSSecretsManager(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsSecretsManagerSecretRotationDestroy,
+		Steps: []resource.TestStep{
+			// Test creating secret rotation resource
+			{
+				Config: testAccAwsSecretsManagerSecretRotationConfig(rName, 7),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsSecretsManagerSecretRotationExists(resourceName, &secret),
+					resource.TestCheckResourceAttr(resourceName, "rotation_enabled", "true"),
+					resource.TestMatchResourceAttr(resourceName, "rotation_lambda_arn", regexp.MustCompile(fmt.Sprintf("^arn:[^:]+:lambda:[^:]+:[^:]+:function:%s-1$", rName))),
+					resource.TestCheckResourceAttr(resourceName, "rotation_rules.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rotation_rules.0.automatically_after_days", "7"),
+				),
+			},
+			// Test updating rotation
+			// We need a valid rotation function for this testing
+			// InvalidRequestException: A previous rotation isn’t complete. That rotation will be reattempted.
+			/*
+				{
+					Config: testAccAwsSecretsManagerSecretConfig_Updated(rName),
+					Check: resource.ComposeTestCheckFunc(
+						testAccCheckAwsSecretsManagerSecretRotationExists(resourceName, &secret),
+						resource.TestCheckResourceAttr(resourceName, "rotation_enabled", "true"),
+						resource.TestMatchResourceAttr(resourceName, "rotation_lambda_arn", regexp.MustCompile(fmt.Sprintf("^arn:[^:]+:lambda:[^:]+:[^:]+:function:%s-2$", rName))),
+					),
+				},
+			*/
+			// Test importing secret rotation
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAwsSecretsManagerSecretRotationDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).secretsmanagerconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_secretsmanager_secret_rotation" {
+			continue
+		}
+
+		input := &secretsmanager.DescribeSecretInput{
+			SecretId: aws.String(rs.Primary.ID),
+		}
+
+		output, err := conn.DescribeSecret(input)
+
+		if err != nil {
+			if isAWSErr(err, secretsmanager.ErrCodeResourceNotFoundException, "") {
+				return nil
```

Comment:
To ensure the for loop runs for all resources, this should use `continue` instead of `return`

```suggestion
				continue
```